### ISSUE TITLE
Add spec for BatchNormTrainingOp

### ISSUE THREAD

**Issue by sdasgup3** [openxla/stablehlo#346]
### Request description

To spec stablehlo.batch_norm_training
